includes/Core/Site_Health/Tags_Placement.php

```diff
@@ -0,0 +1,257 @@
+		add_filter(
+			'googlesitekit_rest_routes',
+			function ( $rest_routes ) {
+				$health_check_routes = $this->get_rest_routes();
+
+				return array_merge( $rest_routes, $health_check_routes );
+			}
+		);
+
+		add_filter(
+			'site_status_tests',
+			function ( $tests ) {
+				global $wp_version;
+
+				if ( version_compare( $wp_version, '5.6', '<' ) ) {
+					$tests['direct']['tag_placement'] = array(
+						'label' => __( 'Tags Placement', 'google-site-kit' ),
+						'test'  => $this->get_method_proxy( 'tags_placement_test' ),
+					);
+
+					return $tests;
+				}
+
+				$tests['async']['tag_placement'] = array(
+					'label'             => __( 'Tags Placement', 'google-site-kit' ),
+					'test'              => rest_url( 'google-site-kit/v1/core/site/data/tags-placement-test' ),
+					'has_rest'          => true,
+					'async_direct_test' => $this->get_method_proxy( 'tags_placement_test' ),
+				);
+
+				return $tests;
+			}
+		);
+	}
+
+	/**
+	 * Gets all REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[] List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		return array(
+			new REST_Route(
+				'core/site/data/tags-placement-test',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => $this->get_method_proxy( 'tags_placement_test' ),
+						'permission_callback' => function () {
+							return current_user_can( Permissions::SETUP );
+						},
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Checks if the modules tags are placed on the website.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Site health status results.
+	 */
+	protected function tags_placement_test() {
+		global $wp_version;
+
+		$result = array(
+			'label'   => __( 'Tags Placement', 'google-site-kit' ),
+			'status'  => 'recommended',
+			'badge'   => array(
+				'label' => __( 'Site Kit', 'google-site-kit' ),
+				'color' => 'blue',
+			),
+			'actions' => '',
+			'test'    => 'tag_placement',
+		);
+
+		if ( version_compare( $wp_version, '5.6', '<' ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'This feature requires WordPress version 5.6 or higher', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$active_modules = $this->get_active_modules();
+		if ( empty( $active_modules ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Tag status not available: AdSense, Tag Manager, and Analytics modules are not connected.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		// Generate random page name that will result in 404 page, to prevent receiving
+		// cached page and target page with smaller content.
+		$random_page = substr(
+			str_shuffle( '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' ),
+			0,
+			10
+		);
+		$response    = wp_remote_get( site_url( $random_page ) ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
```

Comment:
This is more complicated than necessary and introduces a few problems as noted in https://github.com/google/site-kit-wp/issues/5695#issuecomment-1896322675

The main problem is that AdSense tags are intentionally not output on 404 pages, so that check is essentially broken. Second, `site_url` is about the base URL for where `wp-admin` is, not necessarily the front end. Instead, we should check `home_url` and use an added query parameter to serve as a cache buster, just like we do on the client https://github.com/google/site-kit-wp/blob/d952edb61a7604dede836bc3df8c8b8eddc2afbd/assets/js/googlesitekit/datastore/site/html.js#L52-L57

_Note: we should not include the `tagverify` parameter, as this suppresses all SK placed tags_